frontend/lib/src/hooks/useOnInputChange.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dispatch, SetStateAction, useCallback } from "react"
+
+import { isInForm } from "@streamlit/lib/src/util/utils"
+import { ValueWithSource } from "@streamlit/lib/src/hooks/useBasicWidgetState"
+
+type OnInputChangeEventType = {
+  target: { value: string }
+} & Partial<HTMLInputElement>
```

Comment:
**suggestion:** This should better leverage the built-in types:

```suggestion
type OnInputChangeEventType = Pick<
  React.ChangeEvent<HTMLInputElement>,
  "target"
>
```

Reply:
Looking at this a bit further, it seems like we only care about `target.value` in the body of this hook, so perhaps it'd be even simpler to keep this type even more barebones:

```tsx
type OnInputChangeEventType = {
  target: {
    value: HTMLInputElement["value"]
  }
}
```

Reply:
Great, I updated it, thanks!